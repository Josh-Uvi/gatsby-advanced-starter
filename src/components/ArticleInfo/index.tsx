import React from "react";
import { format } from "date-fns";
import { Post } from "../../types";

import {
  TagLink,
  ArticleInfoContainer,
  PostInfo,
  CategoryLink,
  PostInfoCaption,
  TagList,
} from "./style";

type ArticleInfoProps = {
  post: Post;
};

const ArticleInfo = ({ post }: ArticleInfoProps): JSX.Element => {
  const categoryUrl = post.category
    ? `/categories/${post.category}`
    : undefined;

  const publicationDateInfo = `${categoryUrl ? "\u00A0⋅ " : ""}${format(
    post.datePublished,
    "LLL d, y"
  )}`;

  const timeToReadInfo = ` ⋅ ${post.timeToRead} min read`;

  const postInfo = `${publicationDateInfo}${timeToReadInfo}`;

  const tagLinks = post.tags?.slice(0, 2).map((tag) => (
    <TagLink key={tag} to={`/tags/${tag}`}>
      {tag}
    </TagLink>
  ));

  return (
    <ArticleInfoContainer>
      <PostInfo>
        {categoryUrl && (
          <CategoryLink to={categoryUrl}>{post.category}</CategoryLink>
        )}
        <PostInfoCaption>{postInfo}</PostInfoCaption>
      </PostInfo>
      <TagList>{tagLinks}</TagList>
    </ArticleInfoContainer>
  );
};

export default ArticleInfo;
