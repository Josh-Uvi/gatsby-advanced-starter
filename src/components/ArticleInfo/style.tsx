import styled from "styled-components";
import { Link as GatsbyLink } from "gatsby";

import { styles, Caption } from "../../theme";

const Link = styled(GatsbyLink)`
  color: var(--color-primary);
  text-decoration: none;
`;

export const TagLink = styled(Link)`
  ${styles.Overline}

  text-transform: uppercase;

  &::before {
    content: "#";
  }
`;

export const CategoryLink = styled(Link)`
  ${styles.Caption}

  text-transform: capitalize;
`;

export const PostInfoCaption = styled(Caption)`
  color: var(--color-grey-700);
`;

export const PostInfo = styled.div`
  display: flex;
  flex-direction: row;
`;

export const TagList = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
`;

export const ArticleInfoContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;
