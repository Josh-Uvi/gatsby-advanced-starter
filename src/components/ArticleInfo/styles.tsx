import styled from "styled-components";
import { Link as GatsbyLink } from "gatsby";

import { styles, Caption as BaseCaption } from "../../theme";

const Link = styled(GatsbyLink)`
  color: var(--color-primary);
  text-decoration: none;
`;

export const TagLink = styled(Link)`
  ${styles.Overline}

  text-transform: uppercase;

  &::before {
    content: "#";
  }
`;

export const CategoryLink = styled(Link)`
  ${styles.Caption}

  text-transform: capitalize;
`;

export const Caption = styled(BaseCaption)`
  color: var(--color-grey-700);
`;

export const InfoGrid = styled.div`
  display: flex;
  flex-direction: row;
`;

export const TagGrid = styled.div`
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
`;

export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;
