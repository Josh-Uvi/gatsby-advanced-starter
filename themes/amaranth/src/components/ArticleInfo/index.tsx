import React from "react";
import { format } from "date-fns";
import _ from "lodash";

import { Types } from "gatsby-theme-advanced";

import * as S from "./styles";

type ArticleInfoProps = {
  post: Types.Post;
};

const ArticleInfo = ({ post }: ArticleInfoProps): JSX.Element => {
  const categoryUrl = post.category
    ? `/category/${_.kebabCase(post.category.toLowerCase())}`
    : undefined;

  const publicationDate = `${categoryUrl ? "\u00A0⋅ " : ""}${format(
    post.datePublished,
    "LLL d, y"
  )}`;

  const timeToRead = ` ⋅ ${post.timeToRead} min read`;

  const infoCaption = `${publicationDate}${timeToRead}`;

  // Display the first 2 tags
  const tagLinks = post.tags?.slice(0, 2).map((tag) => (
    <S.TagLink key={tag} to={`/tag/${_.kebabCase(tag.toLowerCase())}`}>
      {tag}
    </S.TagLink>
  ));

  return (
    <S.Wrapper>
      <S.InfoGrid>
        {categoryUrl && (
          <S.CategoryLink to={categoryUrl}>{post.category}</S.CategoryLink>
        )}
        <S.Caption>{infoCaption}</S.Caption>
      </S.InfoGrid>
      <S.TagGrid>{tagLinks}</S.TagGrid>
    </S.Wrapper>
  );
};

export default ArticleInfo;
