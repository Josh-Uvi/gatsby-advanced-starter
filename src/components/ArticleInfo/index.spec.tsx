import * as React from "react";
import "jest-styled-components";
import { render } from "@testing-library/react";

import { post } from "../../../test/sampleData";

import ArticleInfo from "./index";

describe("component ArticleInfo", () => {
  it("renders correctly", () => {
    expect.assertions(1);
    const { asFragment } = render(<ArticleInfo post={post} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly without category", () => {
    expect.assertions(1);
    const { asFragment } = render(
      <ArticleInfo post={{ ...post, category: undefined }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
