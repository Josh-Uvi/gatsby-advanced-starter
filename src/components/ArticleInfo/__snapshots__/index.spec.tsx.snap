// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component ArticleInfo renders correctly 1`] = `
<DocumentFragment>
  .c4 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c2 {
  color: var(--color-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-transform: uppercase;
}

.c7::before {
  content: "#";
}

.c3 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: capitalize;
}

.c5 {
  color: var(--color-grey-700);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:564px) {
  .c4 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (min-width:564px) {
  .c7 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    -webkit-letter-spacing: 1.5px;
    -moz-letter-spacing: 1.5px;
    -ms-letter-spacing: 1.5px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }
}

@media (min-width:564px) {
  .c3 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2 c3"
        href="/categories/moar"
      >
        moar
      </a>
      <p
        class="c4 c5"
      >
         ⋅ Mar 1, 2018 ⋅ 3 min read
      </p>
    </div>
    <div
      class="c6"
    >
      <a
        class="c2 c7"
        href="/tags/test"
      >
        test
      </a>
      <a
        class="c2 c7"
        href="/tags/huge"
      >
        huge
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`component ArticleInfo renders correctly without category 1`] = `
<DocumentFragment>
  .c2 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

.c5 {
  color: var(--color-primary);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  font-family: "Fira Sans",sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  text-transform: uppercase;
}

.c6::before {
  content: "#";
}

.c3 {
  color: var(--color-grey-700);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  display: grid;
  grid-auto-flow: column;
  grid-gap: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:564px) {
  .c2 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 16px;
    -webkit-letter-spacing: 0.4px;
    -moz-letter-spacing: 0.4px;
    -ms-letter-spacing: 0.4px;
    letter-spacing: 0.4px;
  }
}

@media (min-width:564px) {
  .c6 {
    font-family: "Fira Sans",sans-serif;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 16px;
    -webkit-letter-spacing: 1.5px;
    -moz-letter-spacing: 1.5px;
    -ms-letter-spacing: 1.5px;
    letter-spacing: 1.5px;
    text-transform: uppercase;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2 c3"
      >
        Mar 1, 2018 ⋅ 3 min read
      </p>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5 c6"
        href="/tags/test"
      >
        test
      </a>
      <a
        class="c5 c6"
        href="/tags/huge"
      >
        huge
      </a>
    </div>
  </div>
</DocumentFragment>
`;
